spras/config/analysis_schema.py

```diff
@@ -0,0 +1,49 @@
+from pydantic import BaseModel, ConfigDict
+
+from spras.config.util_enum import CaseInsensitiveEnum
+
+
+class SummaryAnalysis(BaseModel):
+    include: bool
+
+    model_config = ConfigDict(extra='forbid')
```

Comment:
As a newcomer, it isn't clear why this `CongfigDict` is included at the end of some classes and not others. Is it skipped when a class is a subclass of another?

Reply:
I was looking at `MlLinkage` and `MlMetric` and inferred they are special for some reason because they are used in `MlAnalysis`. But I don't know pydantic, so this was mysterious to me.

Reply:
`MlLinkage` and `MlMetric` are non-pydantic enums: pydantic has [special handling for enums](https://docs.pydantic.dev/2.0/usage/types/enums/) which auto-handles non-conforming values. (We do have a specially defined `CaseInsensitiveEnum` which is why this isn't just over `Enum` instead)

Reply:
Similar to this question, I was also confused until I figured it out myself. I think information like this could be useful to put as a docstring for each of these methods.